src/buildReader.ts

```diff
@@ -338,10 +348,17 @@ export default function buildReader(types: readonly RosMsgDefinition[]): Seriali
   // Since the root message depends on custom types we want those to be defined
   const src = classes.reverse().join("\n\n");
 
+  const fullReader = new MessageReader(types);
+
   // close over our builtin deserializers and builtin size functions
   // eslint-disable-next-line @typescript-eslint/no-implied-eval,no-new-func
-  const wrapFn = new Function("readers", "sizes", `${src}\nreturn __RootMsg;`);
-  const rootMsg = wrapFn.call(undefined, deserializers, builtinSizes) as SerializedMessageReader;
+  const wrapFn = new Function("readers", "sizes", "fullReader", `${src}\nreturn __RootMsg;`);
```

Comment:
nit: would be nice for these variable names to match the ones we pass in to the call below